src/parse.c

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (C) the libgit2 contributors. All rights reserved.
+ *
+ * This file is part of libgit2, distributed under the GNU GPL v2 with
+ * a Linking Exception. For full terms see the included COPYING file.
+ */
+#include "parse.h"
+
+int git_parse_ctx_init(git_parse_ctx *ctx, const char *content, size_t content_len)
+{
+	if (content_len) {
+		if ((ctx->content = git__malloc(content_len)) == NULL) {
+			return -1;
+		}
+
+		memcpy((char *)ctx->content, content, content_len);
+	} else {
+		ctx->content = NULL;
+	}
```

Comment:
Must we copy the contents? We're already read whatever file or buffer into memory.

Reply:
Fixed.